pkg/daemon/update.go

```diff
@@ -779,6 +779,20 @@ func (dn *Daemon) updateFiles(oldConfig, newConfig *mcfgv1.MachineConfig) error
 	return nil
 }
 
+func (dn *Daemon) deleteStaleConfigFile(path string) error {
+	glog.V(2).Infof("Deleting stale config file: %s", path)
+	if err := os.Remove(path); err != nil {
+		newErr := fmt.Errorf("unable to delete %s: %s", path, err)
+		if !os.IsNotExist(err) {
+			return newErr
+		}
+		// otherwise, just warn
+		glog.Warningf("%v", newErr)
+	}
+	glog.Infof("Removed stale file %q", path)
```

Comment:
nit: Technically this should say something like "Stale file %q no longer exists" since it may not have been removed, but that's not a big deal.